### PR TITLE
[ML][Pipelines] parameter_group -> group

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

This PR renamed parameter_group -> group to unify group concept between inputs & outputs

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_group.py [hunk 1]
```diff
@@ -327,21 +327,21 @@ def str_param_pipeline(str_param, group: ParamClass):
...
 
         pipeline_job.inputs.str_param = "test"
         with pytest.raises(Exception) as e:
             pipeline_job.inputs.group = "test"
         assert "'group' is expected to be a parameter group, but got <class 'str'>." in str(e)
 
-    @pytest.mark.skip(reason="Parameter group item .result() is not supported currently.")
-    def test_parameter_group_result(self):
-        @parameter_group
+    @pytest.mark.skip(reason="Input group item .result() is not supported currently.")
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_group.py:338]: BTW what's the current error if we do so on an input in group?

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/constants/_component.py [hunk 2]
```diff
@@ -95,5 +95,5 @@ class IOConstants:
         "string": ["default"],
         "boolean": ["default"],
     }
-    GROUP_ATTR_NAME = "__parameter_group__"
+    GROUP_ATTR_NAME = "__dsl_group__"
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/constants/_component.py:98]: make sure we update the mldesigner._utils recommended azure-ai-ml version, when doing this breaking change

File: sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_group.py (hunk 1 above)
Comment (D-W-) [reply sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_group.py:338]: Calling `result()` on group params will return None, e.g.:

```
        @pipeline()
        def pipeline_with_group(group: ParamClass, int_param: int):
            assert group.str_param.result() is None
```

**Review by brynn-code — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/constants/_component.py (hunk 2 above)
Comment (D-W-) [reply sdk/ml/azure-ai-ml/azure/ai/ml/constants/_component.py:98]: Ideally we should depend on `azure-ai-ml>=1.2.0`?

Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/azure/ai/ml/constants/_component.py:98]: yes.

**Pull request merged**